nixos/modules/services/audio/spotifyd.nix

```diff
@@ -0,0 +1,47 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.spotifyd;
+  spotifydConf = pkgs.writeText "spotifyd.conf" cfg.config;
+in
+{
+  options = {
+    services.spotifyd = {
+      enable = mkEnableOption "spotifyd, a Spotify playing daemon";
+
+      config = mkOption {
+        default = "";
+        type = types.lines;
+        description = ''
+          Configuration for Spotifyd. For syntax and directives, see
+          https://github.com/Spotifyd/spotifyd#Configuration.
+        '';
+      };
+    };
+  };
+
+  config = mkIf cfg.enable {
+    systemd.services.spotifyd = {
+      wantedBy = [ "multi-user.target" ];
+      after = [ "network-online.target" "sound.target" ];
+      description = "spotifyd, a Spotify playing daemon";
+      serviceConfig = {
+        ExecStart = "${pkgs.spotifyd}/bin/spotifyd --no-daemon --cache_path /var/lib/spotifyd --config ${spotifydConf}";
+        Restart = "always";
+        RestartSec = 12;
+        DynamicUser = true;
+        StateDirectory = "spotifyd";
+        User = "spotifyd";
+      };
+    };
+
+    users.users.spotifyd = {
```

Comment:
I think you might be looking for `SupplementaryGroups` instead.

Reply:
d4c88bbc307b0416fc16462a6729cd5960f299e4 fixes this but would you mind explaining these things a bit for me? My understanding of systemd and the NixOS module system is quite limited. I thought it is a good idea to run any independent service under a separate user?

Reply:
By setting `DynamicUser = true;` you are already creating an unprivileged user account independent of any other accounts so you don't need to explicitly declare it. `DynamicUser` is a pretty neat feature of `systemd`!

Reply:
Neat indeed. Thanks for the explanation!